src/dev.ts

```diff
@@ -0,0 +1,279 @@
+ *
+ * This leverages the robust chel serve infrastructure for reliability.
+ */
+class DevEnvironment {
+  private watchers: { close?: () => void }[] = []
+  private serverProcess?: unknown
+  private options: DevOptions
+  private projectRoot: string
+  private metrics: DevMetrics
+  private startTime: Date
+  private contractChanges: ContractChangeEvent[] = []
+
+  constructor (projectRoot: string, options: DevOptions) {
+    this.projectRoot = resolve(projectRoot)
+    this.options = options
+    this.metrics = {
+      contractsWatched: 0,
+      changesDetected: 0,
+      hotReloads: 0,
+      uptime: 0
+    }
+    this.startTime = new Date()
+  }
+
+  // Simple fast non-crypto hash for content (djb2)
+  private hashContent (str: string): string {
+    let h = 5381
+    for (let i = 0; i < str.length; i++) h = ((h << 5) + h) ^ str.charCodeAt(i)
+    // force to unsigned 32-bit and to hex
+    return (h >>> 0).toString(16)
+  }
+
+  async start () {
+    this.printWelcomeBanner()
+    await this.startServer()
+    await this.startContractWatching()
+    await this.startInteractiveMode()
+  }
+
+  private async startServer () {
+    console.log(colors.blue('🚀 Starting development server with hot reload...'))
+
+    // Start the server in the background using our robust serve infrastructure
+    const serverArgs = [
+      '--dp', String(this.options.dashboardPort || 3000),
+      '--port', String(this.options.port || 8000),
+      '--db-type', 'mem', // Use memory for faster dev iterations
+      this.projectRoot
+    ]
+
+    try {
+      // Start the serve function in the background
+      // Don't await it here as it runs indefinitely
+      this.serverProcess = serve(serverArgs).catch(error => {
+        console.error(colors.red('❌ Server error:'), error)
+        process.exit(1)
+      })
+
+      // Give the server a moment to start
+      await new Promise(resolve => setTimeout(resolve, 2000))
+
+      console.log(colors.green('✅ Development server started'))
+      console.log(colors.gray(`   Dashboard: http://localhost:${this.options.dashboardPort || 3000}`))
+      console.log(colors.gray(`   Application: http://localhost:${this.options.port || 8000}`))
+    } catch (error) {
+      console.error(colors.red('❌ Failed to start development server:'), error)
+      throw error
+    }
+  }
+
+  private printWelcomeBanner () {
+    console.log(colors.cyan('╔══════════════════════════════════════════════════════════════╗'))
+    console.log(colors.cyan('║') + colors.bold(colors.white('               🧪 CHEL LIVE DEVELOPMENT                     ')) + colors.cyan('║'))
+    console.log(colors.cyan('║') + colors.white('           Live-testing with Hot Reload & Interaction        ') + colors.cyan('║'))
+    console.log(colors.cyan('╚══════════════════════════════════════════════════════════════╝'))
+    console.log()
+    console.log(colors.gray('Similar to grunt dev - provides live-testing environment'))
```

Comment:
No need to mention `grunt dev` here, as that's irrelevant and likely confusing information (from the perspective of someone using `chel dev` who hasn't worked on GI before).